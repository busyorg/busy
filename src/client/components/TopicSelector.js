import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import SortSelector from './SortSelector/SortSelector';
import Topic from './Button/Topic';
import './TopicSelector.less';

const TopicSelector = ({ sort, isSingle, topics, onTopicClose, onSortChange }) => (
  <div className="TopicSelector">
    <div className="TopicSelector__topics">
      {topics &&
        topics.map(topic => (
          <Topic key={topic} closable={!isSingle} name={topic} onClose={onTopicClose} />
        ))}
    </div>
    <SortSelector sort={sort} onChange={onSortChange}>
      <SortSelector.Item key="trending">
        <FormattedMessage id="sort_trending" defaultMessage="Trending" />
      </SortSelector.Item>
      <SortSelector.Item key="created">
        <FormattedMessage id="sort_created" defaultMessage="Created" />
      </SortSelector.Item>
      <SortSelector.Item key="active">
        <FormattedMessage id="sort_active" defaultMessage="Active" />
      </SortSelector.Item>
      <SortSelector.Item key="hot">
        <FormattedMessage id="sort_hot" defaultMessage="Hot" />
      </SortSelector.Item>
    </SortSelector>
  </div>
);

TopicSelector.propTypes = {
  sort: PropTypes.string,
  isSingle: PropTypes.bool,
  topics: PropTypes.arrayOf(PropTypes.string),
  onTopicClose: PropTypes.func,
  onSortChange: PropTypes.func,
};

TopicSelector.defaultProps = {
  sort: 'trending',
  isSingle: true,
  topics: [],
  onTopicClose: () => {},
  onSortChange: () => {},
};

export default TopicSelector;
