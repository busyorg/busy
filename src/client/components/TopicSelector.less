@import (reference) '../styles/modules/variables.less';

.TopicSelector {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;
  padding: 0 @mobile-padding;

  @media @small {
    padding: 0;
  }

  &__topics {
    & > a {
      &:not(:nth-child(1)) {
        margin-left: 4px;
      }
    }
  }
}
