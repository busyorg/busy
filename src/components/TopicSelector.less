@import (reference) "../styles/modules/variables.less";

.TopicSelector {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;
  padding: 0 @mobile-padding;

  @media @small {
    padding: 0;
  }

  &__topics {
    & > a {
      &:not(:nth-child(1)) {
        margin-left: 4px;
      }
    }
  }

  &__sort {
    color: #99aab5;
    text-align: right;

    &__title {
      display: inline-block;
      white-space: nowrap;
    }

    &__current {
      cursor: pointer;
      margin-left: 8px;
      font-weight: 500;
      white-space: nowrap;

      & > .iconfont {
        margin-left: 4px;
        font-size: 16px !important;
        vertical-align: -1px;
      }
    }
  }
}
