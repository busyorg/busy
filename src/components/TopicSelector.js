import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Popover } from 'antd';
import PopoverMenu, { PopoverMenuItem } from './PopoverMenu/PopoverMenu';
import Topic from './Button/Topic';
import './TopicSelector.less';

class TopicSelector extends React.Component {
  static propTypes = {
    sort: PropTypes.string,
    isSingle: PropTypes.bool,
    bold: PropTypes.bool,
    topics: PropTypes.arrayOf(PropTypes.string),
    onTopicClose: PropTypes.func,
    onSortChange: PropTypes.func,
  };

  static defaultProps = {
    sort: 'trending',
    isSingle: true,
    bold: true,
    topics: [],
    onTopicClose: () => {},
    onSortChange: () => {},
  };

  constructor(props) {
    super(props);
    this.state = {
      popoverVisible: false,
    };
  }

  onSelect = (key) => {
    this.setState({ popoverVisible: false }, () => {
      this.props.onSortChange(key);
    });
  };

  onVisibleChange = visible => this.setState({ popoverVisible: visible });

  render() {
    const { sort, topics, isSingle, bold, onTopicClose } = this.props;
    const { popoverVisible } = this.state;

    return (
      <div className="TopicSelector">
        <div className="TopicSelector__topics">
          {topics &&
            topics.map(topic =>
              <Topic key={topic} closable={!isSingle} name={topic} onClose={onTopicClose} />,
            )}
        </div>
        <div className="TopicSelector__sort">
          <span className="TopicSelector__sort__title">
            <FormattedMessage id="sort_by" defaultMessage="Sort by" />
          </span>
          <Popover
            trigger="click"
            visible={popoverVisible}
            onVisibleChange={this.onVisibleChange}
            content={
              <PopoverMenu bold={bold} onSelect={this.onSelect}>
                <PopoverMenuItem key="trending">
                  <FormattedMessage id="sort_trending" defaultMessage="Trending" />
                </PopoverMenuItem>
                <PopoverMenuItem key="created">
                  <FormattedMessage id="sort_created" defaultMessage="Created" />
                </PopoverMenuItem>
                <PopoverMenuItem key="active">
                  <FormattedMessage id="sort_active" defaultMessage="Active" />
                </PopoverMenuItem>
                <PopoverMenuItem key="hot">
                  <FormattedMessage id="sort_hot" defaultMessage="Hot" />
                </PopoverMenuItem>
                <PopoverMenuItem key="promoted">
                  <FormattedMessage id="sort_promoted" defaultMessage="Promoted" />
                </PopoverMenuItem>
              </PopoverMenu>
            }
          >
            <span className="TopicSelector__sort__current">
              <FormattedMessage
                id={`sort_${sort}`}
                defaultMessage={sort.charAt(0).toUpperCase() + sort.slice(1)}
              />
              <i className="iconfont icon-unfold" />
            </span>
          </Popover>
        </div>
      </div>
    );
  }
}

export default TopicSelector;
