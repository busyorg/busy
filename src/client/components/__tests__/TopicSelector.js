import React from 'react';
import { shallow } from 'enzyme';
import TopicSelector from '../TopicSelector';

describe('<TopicSelector />', () => {
  it('renders without exploding', () => {
    const props = {
      sort: 'active',
      isSingle: true,
      bold: true,
      topics: ['busy', 'introduceyourself', 'steemfest'],
    };
    const wrapper = shallow(<TopicSelector {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
