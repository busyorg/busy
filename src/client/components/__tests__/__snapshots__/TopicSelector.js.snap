// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicSelector /> renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopicSelector
    bold={true}
    isSingle={true}
    onSortChange={[Function]}
    onTopicClose={[Function]}
    sort="active"
    topics={
      Array [
        "busy",
        "introduceyourself",
        "steemfest",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="TopicSelector__topics"
        >
          <Topic
            closable={false}
            favorite={false}
            name="busy"
            onClose={[Function]}
          />
          <Topic
            closable={false}
            favorite={false}
            name="introduceyourself"
            onClose={[Function]}
          />
          <Topic
            closable={false}
            favorite={false}
            name="steemfest"
            onClose={[Function]}
          />
        </div>,
        <SortSelector
          onChange={[Function]}
          sort="active"
        >
          <PopoverMenuItem
            bold={true}
            disabled={false}
            fullScreenHidden={false}
            itemKey=""
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Trending"
              id="sort_trending"
              values={Object {}}
            />
          </PopoverMenuItem>
          <PopoverMenuItem
            bold={true}
            disabled={false}
            fullScreenHidden={false}
            itemKey=""
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Created"
              id="sort_created"
              values={Object {}}
            />
          </PopoverMenuItem>
          <PopoverMenuItem
            bold={true}
            disabled={false}
            fullScreenHidden={false}
            itemKey=""
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Active"
              id="sort_active"
              values={Object {}}
            />
          </PopoverMenuItem>
          <PopoverMenuItem
            bold={true}
            disabled={false}
            fullScreenHidden={false}
            itemKey=""
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Hot"
              id="sort_hot"
              values={Object {}}
            />
          </PopoverMenuItem>
        </SortSelector>,
      ],
      "className": "TopicSelector",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Topic
              closable={false}
              favorite={false}
              name="busy"
              onClose={[Function]}
            />,
            <Topic
              closable={false}
              favorite={false}
              name="introduceyourself"
              onClose={[Function]}
            />,
            <Topic
              closable={false}
              favorite={false}
              name="steemfest"
              onClose={[Function]}
            />,
          ],
          "className": "TopicSelector__topics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "busy",
            "nodeType": "class",
            "props": Object {
              "closable": false,
              "favorite": false,
              "name": "busy",
              "onClose": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "introduceyourself",
            "nodeType": "class",
            "props": Object {
              "closable": false,
              "favorite": false,
              "name": "introduceyourself",
              "onClose": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "steemfest",
            "nodeType": "class",
            "props": Object {
              "closable": false,
              "favorite": false,
              "name": "steemfest",
              "onClose": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Trending"
                id="sort_trending"
                values={Object {}}
              />
            </PopoverMenuItem>,
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Created"
                id="sort_created"
                values={Object {}}
              />
            </PopoverMenuItem>,
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Active"
                id="sort_active"
                values={Object {}}
              />
            </PopoverMenuItem>,
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Hot"
                id="sort_hot"
                values={Object {}}
              />
            </PopoverMenuItem>,
          ],
          "onChange": [Function],
          "sort": "active",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "trending",
            "nodeType": "function",
            "props": Object {
              "bold": true,
              "children": <FormattedMessage
                defaultMessage="Trending"
                id="sort_trending"
                values={Object {}}
              />,
              "disabled": false,
              "fullScreenHidden": false,
              "itemKey": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Trending",
                "id": "sort_trending",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "created",
            "nodeType": "function",
            "props": Object {
              "bold": true,
              "children": <FormattedMessage
                defaultMessage="Created"
                id="sort_created"
                values={Object {}}
              />,
              "disabled": false,
              "fullScreenHidden": false,
              "itemKey": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Created",
                "id": "sort_created",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "active",
            "nodeType": "function",
            "props": Object {
              "bold": true,
              "children": <FormattedMessage
                defaultMessage="Active"
                id="sort_active"
                values={Object {}}
              />,
              "disabled": false,
              "fullScreenHidden": false,
              "itemKey": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Active",
                "id": "sort_active",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "hot",
            "nodeType": "function",
            "props": Object {
              "bold": true,
              "children": <FormattedMessage
                defaultMessage="Hot"
                id="sort_hot"
                values={Object {}}
              />,
              "disabled": false,
              "fullScreenHidden": false,
              "itemKey": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Hot",
                "id": "sort_hot",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="TopicSelector__topics"
          >
            <Topic
              closable={false}
              favorite={false}
              name="busy"
              onClose={[Function]}
            />
            <Topic
              closable={false}
              favorite={false}
              name="introduceyourself"
              onClose={[Function]}
            />
            <Topic
              closable={false}
              favorite={false}
              name="steemfest"
              onClose={[Function]}
            />
          </div>,
          <SortSelector
            onChange={[Function]}
            sort="active"
          >
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Trending"
                id="sort_trending"
                values={Object {}}
              />
            </PopoverMenuItem>
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Created"
                id="sort_created"
                values={Object {}}
              />
            </PopoverMenuItem>
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Active"
                id="sort_active"
                values={Object {}}
              />
            </PopoverMenuItem>
            <PopoverMenuItem
              bold={true}
              disabled={false}
              fullScreenHidden={false}
              itemKey=""
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Hot"
                id="sort_hot"
                values={Object {}}
              />
            </PopoverMenuItem>
          </SortSelector>,
        ],
        "className": "TopicSelector",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Topic
                closable={false}
                favorite={false}
                name="busy"
                onClose={[Function]}
              />,
              <Topic
                closable={false}
                favorite={false}
                name="introduceyourself"
                onClose={[Function]}
              />,
              <Topic
                closable={false}
                favorite={false}
                name="steemfest"
                onClose={[Function]}
              />,
            ],
            "className": "TopicSelector__topics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "busy",
              "nodeType": "class",
              "props": Object {
                "closable": false,
                "favorite": false,
                "name": "busy",
                "onClose": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "introduceyourself",
              "nodeType": "class",
              "props": Object {
                "closable": false,
                "favorite": false,
                "name": "introduceyourself",
                "onClose": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "steemfest",
              "nodeType": "class",
              "props": Object {
                "closable": false,
                "favorite": false,
                "name": "steemfest",
                "onClose": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PopoverMenuItem
                bold={true}
                disabled={false}
                fullScreenHidden={false}
                itemKey=""
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Trending"
                  id="sort_trending"
                  values={Object {}}
                />
              </PopoverMenuItem>,
              <PopoverMenuItem
                bold={true}
                disabled={false}
                fullScreenHidden={false}
                itemKey=""
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Created"
                  id="sort_created"
                  values={Object {}}
                />
              </PopoverMenuItem>,
              <PopoverMenuItem
                bold={true}
                disabled={false}
                fullScreenHidden={false}
                itemKey=""
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Active"
                  id="sort_active"
                  values={Object {}}
                />
              </PopoverMenuItem>,
              <PopoverMenuItem
                bold={true}
                disabled={false}
                fullScreenHidden={false}
                itemKey=""
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Hot"
                  id="sort_hot"
                  values={Object {}}
                />
              </PopoverMenuItem>,
            ],
            "onChange": [Function],
            "sort": "active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "trending",
              "nodeType": "function",
              "props": Object {
                "bold": true,
                "children": <FormattedMessage
                  defaultMessage="Trending"
                  id="sort_trending"
                  values={Object {}}
                />,
                "disabled": false,
                "fullScreenHidden": false,
                "itemKey": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Trending",
                  "id": "sort_trending",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "created",
              "nodeType": "function",
              "props": Object {
                "bold": true,
                "children": <FormattedMessage
                  defaultMessage="Created"
                  id="sort_created"
                  values={Object {}}
                />,
                "disabled": false,
                "fullScreenHidden": false,
                "itemKey": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Created",
                  "id": "sort_created",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "active",
              "nodeType": "function",
              "props": Object {
                "bold": true,
                "children": <FormattedMessage
                  defaultMessage="Active"
                  id="sort_active"
                  values={Object {}}
                />,
                "disabled": false,
                "fullScreenHidden": false,
                "itemKey": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Active",
                  "id": "sort_active",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "hot",
              "nodeType": "function",
              "props": Object {
                "bold": true,
                "children": <FormattedMessage
                  defaultMessage="Hot"
                  id="sort_hot"
                  values={Object {}}
                />,
                "disabled": false,
                "fullScreenHidden": false,
                "itemKey": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Hot",
                  "id": "sort_hot",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
